// vision-camera-qrcode-scanner/lib/module/index.js
/**
 * @see https://developers.google.com/android/reference/com/google/mlkit/vision/barcode/Barcode.BarcodeValueType
 */
export let QrCodeType;
/**
 * @see https://developers.google.com/android/reference/com/google/mlkit/vision/barcode/Barcode.Address.AddressType
 */

(function (QrCodeType) {
  QrCodeType[QrCodeType["UNKNOWN"] = 0] = "UNKNOWN";
  QrCodeType[QrCodeType["CONTACT_INFO"] = 1] = "CONTACT_INFO";
  QrCodeType[QrCodeType["EMAIL"] = 2] = "EMAIL";
  QrCodeType[QrCodeType["ISBN"] = 3] = "ISBN";
  QrCodeType[QrCodeType["PHONE"] = 4] = "PHONE";
  QrCodeType[QrCodeType["PRODUCT"] = 5] = "PRODUCT";
  QrCodeType[QrCodeType["SMS"] = 6] = "SMS";
  QrCodeType[QrCodeType["TEXT"] = 7] = "TEXT";
  QrCodeType[QrCodeType["URL"] = 8] = "URL";
  QrCodeType[QrCodeType["WIFI"] = 9] = "WIFI";
  QrCodeType[QrCodeType["GEO"] = 10] = "GEO";
  QrCodeType[QrCodeType["CALENDAR_EVENT"] = 11] = "CALENDAR_EVENT";
  QrCodeType[QrCodeType["DRIVER_LICENSE"] = 12] = "DRIVER_LICENSE";
})(QrCodeType || (QrCodeType = {}));

export let QrCodeAddressType;
/**
 * @see https://developers.google.com/android/reference/com/google/mlkit/vision/barcode/Barcode.Address
 */

(function (QrCodeAddressType) {
  QrCodeAddressType[QrCodeAddressType["UNKNOWN"] = 0] = "UNKNOWN";
  QrCodeAddressType[QrCodeAddressType["WORK"] = 1] = "WORK";
  QrCodeAddressType[QrCodeAddressType["HOME"] = 2] = "HOME";
})(QrCodeAddressType || (QrCodeAddressType = {}));

/**
 * @see https://developers.google.com/android/reference/com/google/mlkit/vision/barcode/Barcode.Email.FormatType
 */
export let QrCodeEmailType;
/**
 * @see https://developers.google.com/android/reference/com/google/mlkit/vision/barcode/Barcode.Email
 */

(function (QrCodeEmailType) {
  QrCodeEmailType[QrCodeEmailType["UNKNOWN"] = 0] = "UNKNOWN";
  QrCodeEmailType[QrCodeEmailType["WORK"] = 1] = "WORK";
  QrCodeEmailType[QrCodeEmailType["HOME"] = 2] = "HOME";
})(QrCodeEmailType || (QrCodeEmailType = {}));

/**
 * @see https://developers.google.com/android/reference/com/google/mlkit/vision/barcode/Barcode.Phone.FormatType
 */
export let QrCodePhoneType;
/**
 * @see https://developers.google.com/android/reference/com/google/mlkit/vision/barcode/Barcode.Phone
 */

(function (QrCodePhoneType) {
  QrCodePhoneType[QrCodePhoneType["UNKNOWN"] = 0] = "UNKNOWN";
  QrCodePhoneType[QrCodePhoneType["WORK"] = 1] = "WORK";
  QrCodePhoneType[QrCodePhoneType["HOME"] = 2] = "HOME";
  QrCodePhoneType[QrCodePhoneType["FAX"] = 3] = "FAX";
  QrCodePhoneType[QrCodePhoneType["MOBILE"] = 4] = "MOBILE";
})(QrCodePhoneType || (QrCodePhoneType = {}));

/**
 * @see https://developers.google.com/android/reference/com/google/mlkit/vision/barcode/Barcode.WiFi.EncryptionType
 */
export let QrCodeWifiType;
/**
 * @see https://developers.google.com/android/reference/com/google/mlkit/vision/barcode/Barcode.WiFi
 */

(function (QrCodeWifiType) {
  QrCodeWifiType[QrCodeWifiType["OPEN"] = 1] = "OPEN";
  QrCodeWifiType[QrCodeWifiType["WPA"] = 2] = "WPA";
  QrCodeWifiType[QrCodeWifiType["WEP"] = 3] = "WEP";
})(QrCodeWifiType || (QrCodeWifiType = {}));

/**
 * Scans QR codes.
 */
export function scanQRCodes(frame) {
  'worklet'; // @ts-ignore
  // eslint-disable-next-line no-undef

  return __scanQRCodes(frame);
}
